ui/packages/atlasmap/src/UI/dialogs/ParametersDialog.tsx

```diff
@@ -0,0 +1,178 @@
+        newParameter.hidden = false;
+        break;
+      }
+    }
+    setAddParameter(initialAddParameter);
+    setParameters(newParameters);
+  }, [parameters, addParameter]);
+
+  const handleRemoveParameter = useCallback((name) => {
+    const newParameters = [...parameters];
+    for (const newParameter of newParameters) {
+      if (newParameter.name === name) {
+        newParameter.hidden = true;
+        break;
+      }
+    }
+    setParameters(newParameters);
+  }, [parameters]);
+
+  const handleOnChangeParameter = useCallback((value, event) => {
+    const newParameters = [...parameters];
+    for (const newParameter of newParameters) {
+      if (newParameter.name === event.target.name) {
+        newParameter.value = value;
+        break;
+      }
+    }
+    setParameters(newParameters);
+  }, [parameters]);
+
+  const hiddenParameters = (): IParameter[] => {
+    return parameters.filter(parameter => parameter.hidden);
+  };
+
+  const visibleParameters = (): IParameter[] => {
+    return parameters.filter(parameter => !parameter.hidden);
+  };
+
+  // make sure to resync the internal state to the values passed in as props
+  useEffect(reset, [reset]);
+
+  return (
+    <ConfirmationDialog
+      title={title}
+      onCancel={handleOnCancel}
+      onConfirm={parameters.length > 0 ? handleOnConfirm : undefined}
+      isOpen={isOpen}
+    >
+      <Form>
+        {visibleParameters().map((parameter, index) => (
+          <FormGroup key={index} label={parameter.name} fieldId={parameter.name}>
+            <Split>
+              <SplitItem>
+                {(parameter.options && parameter.options.length > 0) ?
+                  (<FormSelect
+                      value={parameter.value}
+                      onChange={(value, event) => handleOnChangeParameter(value, event)}
+                      id={parameter.name}
+                      name={parameter.name}
+                      data-testid={parameter.name + "-parameter-form-select"}
+                  >
+                    {parameter.options.map(({ label, value }, idx) => (
+                        <FormSelectOption key={idx} value={value} label={label} />
+                    ))}
+                  </FormSelect>)
+                :
+                  (<TextInput
+                    value={parameter.value}
+                    onChange={(value, event) => handleOnChangeParameter(value, event)}
+                    id={parameter.name}
+                    name={parameter.name}
+                    data-testid={parameter.name + "-parameter-text-input"}
+                  />)
+                }
+              </SplitItem>
+              {parameter.hideable &&
+                <SplitItem>
+                  <Button variant="plain" aria-label="Action" onClick={(_event) => handleRemoveParameter(parameter.name)}>
+                    <TimesIcon />
+                  </Button>
+                </SplitItem>
+              }
+            </Split>
+          </FormGroup>
```

Comment:
This block is a great candidate to become a component on its own. It doesn't have to be in a new file since it's used only here, but will make reading the code easier.

Reply:
Automated refactoring doesn't do the trick so I left it as is.